bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: embed OpenAPI contents using Go embed

## Description

This PR replaces the usage of `go-bindata` to use native Go embedding for the public OpenAPI files of Managed Kafka and Managed Connectors.

Since Go 1.16 Go added native file/directory embedding. More information about Go native embedding can be found in https://pkg.go.dev/embed.

When running `go build` to build the fleet manager binary, Go automatically embeds the current contents into the binary.

This allows us to stop having the additional `go-bindata` dependency and tooling to perform file embedding. It also simplifies complexity of the code and file embedding.

Some remarks:
* The `.go` files containing the embedded contents are in the same directory as the openapi specification files. The reason for that is that in native Go embedding the `.go` files that contain embed references can only reference files in their same directory or in subdirectories but not in parent directories. An alternative could have been to move the openAPI specification files to a `openapi/specifications` subdirectory but I was not sure someone else (KAS fleet manager SDK/CLI?) depended on this hence why in the implementation the Go files are in the same directory.
* References and usage to go-bindata have been completely removed. The binary is no longer installed and the tool is not used anymore, nor the files it used to generate.






## Verification Steps

1. Build the fleet manager binary with `make binary`
1. Run the fleet manager
1. Request the openapi content to the fleet manager using its API: `curl -H "Authorization: Bearer $(ocm token)" 127.0.0.1:8000/api/kafkas_mgmt/v1/openapi | jq .`
1. Verify that the returned response is the OpenAPI content defined in openapi/kas-fleet-manager.yaml in JSON format

The same test has to be performed for the connectors fleet manager. A workaround way to test it from here is to modify the `cmd/kas-fleet-manager/main.go` file doing the following changes:
*  Import the `github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/connector` package
*  In that file, when the the environment is created replace `kafka.ConfigProviders()` by `connector.ConfigProviders(false)`.

Then repeat all the previously mentioned steps but now request the OpenAPI file for connectors by running:
`curl -H "Authorization: Bearer $(ocm token)" 127.0.0.1:8000/api/connector_mgmt/v1/openapi | jq .`
Finally verify that the returned response is the OpenAPI content defined in openapi/connector_mgmt.yaml.



## Checklist (Definition of Done)
